Do not crash and return 500 error when requested file doesn't exist on disk

```
[2025-03-28 15:56:12 +0100] [8] [INFO] Application startup complete.
thread 'main' panicked at library/std/src/io/stdio.rs:1118:9:
failed printing to stdout: Broken pipe (os error 32)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Error: Os { code: 2, kind: NotFound, message: "No such file or directory" }
Error: Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2025-03-31 07:58:34 +0200] [8] [ERROR] Exception in ASGI application
Traceback (most recent call last):
  File "/venv/lib/python3.11/site-packages/uvicorn/protocols/http/h11_impl.py", line 429, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/venv/lib/python3.11/site-packages/starlette/applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 187, in __call__
    raise exc
  File "/venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 165, in __call__
    await self.app(scope, receive, _send)
  File "/venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "/venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "/venv/lib/python3.11/site-packages/starlette/routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/venv/lib/python3.11/site-packages/starlette/routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "/venv/lib/python3.11/site-packages/starlette/routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "/venv/lib/python3.11/site-packages/starlette/routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "/venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "/venv/lib/python3.11/site-packages/starlette/routing.py", line 73, in app
    response = await f(request)
               ^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/fastapi/routing.py", line 301, in app
    raw_response = await run_endpoint_function(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/fastapi/routing.py", line 212, in run_endpoint_function
    return await dependant.call(**values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/src/chanjo2/endpoints/overview.py", line 82, in overview
    overview_content: dict = get_report_data(
                             ^^^^^^^^^^^^^^^^
  File "/home/worker/app/src/chanjo2/meta/handle_report_contents.py", line 114, in get_report_data
    data["sex_rows"] = get_report_sex_rows(samples=query.samples)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/src/chanjo2/meta/handle_report_contents.py", line 187, in get_report_sex_rows
    sample_sex_metrics: Dict = get_samples_sex_metrics(
                               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/src/chanjo2/meta/handle_d4.py", line 186, in get_samples_sex_metrics
    sex_chroms_coverage: List[Tuple[str, float]] = get_d4tools_chromosome_mean_coverage(
                                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/src/chanjo2/meta/handle_coverage_stats.py", line 70, in get_d4tools_chromosome_mean_coverage
    chromosomes_stats_mean_cmd: List[str] = subprocess.check_output(
                                            ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/subprocess.py", line 466, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['d4tools', 'stat', '-smean', '/Users/chiararasi/Documents/work/GITs/scout/scout/demo/ADM1059A2.d4']' returned non-zero exit status 1.
[hiseq.clinical@cg-services-stage ~]$
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,7 @@
+## [unreleased]
+### Changed
+- Introduced a validation to make sure d4 files are existing on disk when creating reports
+
 ## [3.4]
 ### Changed
 - General library updates

--- a/src/chanjo2/models/pydantic_models.py
+++ b/src/chanjo2/models/pydantic_models.py
@@ -2,6 +2,7 @@ import json
 import os
 from datetime import datetime
 from enum import Enum
+from os.path import isfile
 from typing import Dict, List, Optional, Union
 
 from pydantic import BaseModel, ConfigDict, Field, field_validator, model_validator
@@ -11,6 +12,7 @@ from chanjo2.constants import (
     DEFAULT_COMPLETENESS_LEVELS,
     DEFAULT_COVERAGE_LEVEL,
     GENE_LISTS_NOT_SUPPORTED_MSG,
+    WRONG_COVERAGE_FILE_MSG,
 )
 
 
@@ -169,6 +171,12 @@ class ReportQuerySample(BaseModel):
     coverage_file_path: Optional[str] = None
     case_name: Optional[str] = None
     analysis_date: Optional[datetime] = datetime.now()
+
+    @field_validator("coverage_file_path", mode="after")
+    def coverage_file_path_validator(cls, coverage_file_path):
+        if isfile(coverage_file_path):
+            return coverage_file_path
+        raise ValueError(WRONG_COVERAGE_FILE_MSG)
 
 
 class ReportQuery(BaseModel):

--- a/tests/src/chanjo2/endpoints/test_overview.py
+++ b/tests/src/chanjo2/endpoints/test_overview.py
@@ -4,8 +4,13 @@ from fastapi import status
 from fastapi.testclient import TestClient
 from requests.models import Response
 
-from chanjo2.constants import BUILD_37, BUILD_38, DEFAULT_COMPLETENESS_LEVELS
-from chanjo2.demo import DEMO_COVERAGE_QUERY_FORM
+from chanjo2.constants import (
+    BUILD_37,
+    BUILD_38,
+    DEFAULT_COMPLETENESS_LEVELS,
+    WRONG_COVERAGE_FILE_MSG,
+)
+from chanjo2.demo import DEMO_COVERAGE_QUERY_FORM, d4_demo_path
 
 
 def test_demo_overview(client: TestClient, endpoints: Type):
@@ -35,6 +40,25 @@ def test_overview_form_data(client: TestClient, endpoints: Type):
 
     # And return an HTML page
     assert response.template.name == "overview.html"
+
+
+def test_overview_form_data_d4_not_found(client: TestClient, endpoints: Type):
+    """Test the endpoint that creates the genes coverage overview page by providing a sample with a d4 file not found on disk."""
+
+    # GIVEN a query with application/x-www-form-urlencoded data and a sample with no valid d4 file on disk
+    data = DEMO_COVERAGE_QUERY_FORM.copy()
+    data["samples"] = data["samples"].replace(d4_demo_path, "foo")
+
+    response: Response = client.post(
+        endpoints.OVERVIEW,
+        data=data,
+        headers={"Content-Type": "application/x-www-form-urlencoded"},
+    )
+    # Then the request should be returning error
+    assert response.status_code == status.HTTP_422_UNPROCESSABLE_ENTITY
+
+    # And return the expected validation error
+    assert WRONG_COVERAGE_FILE_MSG in str(response.json())
 
 
 def test_gene_overview(

